src/main/scala/Head.scala

```diff
@@ -1,5 +1,27 @@
 package reinvent
 
 object Head {
-  def apply(input: List[Int]): Int = ???
+  def apply[A](input: List[A]): A = input match {
+    case Nil => throw new RuntimeException("Head of empty list")
```

Comment:
You might want to use a more specific exception. For example the head in prelude will return 'java.util.NoSuchElementException'.